subnet/subnet_contract.go

```diff
@@ -3,27 +3,57 @@ package subnet
 import (
 	"github.com/nspcc-dev/neo-go/pkg/interop"
 	"github.com/nspcc-dev/neo-go/pkg/interop/contract"
+	"github.com/nspcc-dev/neo-go/pkg/interop/iterator"
 	"github.com/nspcc-dev/neo-go/pkg/interop/native/management"
 	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
 	"github.com/nspcc-dev/neo-go/pkg/interop/storage"
 	"github.com/nspcc-dev/neofs-contract/common"
 )
 
 const (
-	// ErrInvalidSubnetID is thrown when subnet id is not a slice of 4 bytes.
+	// ErrInvalidSubnetID is thrown when subnet id is not a slice of 5 bytes.
 	ErrInvalidSubnetID = "invalid subnet ID"
-	// ErrInvalidOwner is thrown when owner has invalid format.
-	ErrInvalidOwner = "invalid owner"
+	// ErrInvalidGroupID is thrown when group id is not a slice of 5 bytes.
+	ErrInvalidGroupID = "invalid group ID"
 	// ErrAlreadyExists is thrown when id already exists.
 	ErrAlreadyExists = "subnet id already exists"
 	// ErrNotExist is thrown when id doesn't exist.
 	ErrNotExist = "subnet id doesn't exist"
+	// ErrInvalidOwner is thrown when owner has invalid format.
+	ErrInvalidOwner = "invalid owner"
+	// ErrInvalidAdmin is thrown when admin has invalid format.
+	ErrInvalidAdmin = "invalid administrator"
+	// ErrInvalidNode is thrown when node has invalid format.
+	ErrInvalidNode = "invalid node key"
+	// ErrNodeAdmNotExist is thrown when node admin is not found.
+	ErrNodeAdmNotExist = "node admin not found"
+	// ErrClientAdmNotExist is thrown when client admin is not found.
+	ErrClientAdmNotExist = "client admin not found"
+	// ErrNodeNotExist is thrown when node is not found.
+	ErrNodeNotExist = "node not found"
+	// ErrUserNotExist is thrown when user is not found.
+	ErrUserNotExist = "user not found"
+	// ErrAccessDenied is thrown when operation is denied for caller.
+	ErrAccessDenied = "access denied"
 
-	ownerPrefix       = 'o'
+	errCheckWitnessFailed = "owner witness check failed"
+)
+
+const (
+	nodeAdminPrefix   = 'a'
 	infoPrefix        = 'i'
+	clientAdminPrefix = 'm'
+	nodePrefix        = 'n'
+	ownerPrefix       = 'o'
+	userPrefix        = 'u'
 	notaryDisabledKey = 'z'
 )
 
+const (
+	subnetIDSize = 5
+	groupIDSize  = 5
```

Comment:
there will be proto but not in the API repository. it is not gonna be used by nodes or IR. it will be used by admins. so I guess that it could be somewhere in the `neofs-adm`